java-frontend/src/main/java/org/sonar/plugins/java/api/tree/OpensDirectiveTree.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.java.api.tree;
+
+import com.google.common.annotations.Beta;
+
+import javax.annotation.Nullable;
+
+/**
+ * The 'opens' directive from java 9 module directives
+ *
+ * JLS9 - §7.7.2
+ *
+ * <pre>
+ *   opens {@link #packageName()} ;
+ *   opens {@link #packageName()} to {@link #moduleNames()} ;
+ * </pre>
+ *
+ * @since Java 9
+ */
+@Beta
+public interface OpensDirectiveTree extends ModuleDirectiveTree {
+
+  ExpressionTree packageName();
```

Comment:
I disagree. The current {{packageName}} of the {{PackageDeclarationTree}} is already an {{ExpressionTree}}. I prefer to stay coherent and use the same approach when parsing package names. It would also simplify the way we compare package name (trees will be the same).